CharlieBackend.Business/Services/LessonService.cs

```diff
@@ -154,11 +158,11 @@ public async Task<LessonDto> UpdateLessonAsync(long id, UpdateLessonDto lessonMo
             }
         }
 
-        public async Task<IList<StudentLessonDto>> GetStudentLessonsAsync(long studentId)
+        public async Task<Result<IList<StudentLessonDto>>> GetStudentLessonsAsync(long studentId)
         {
             var studentLessonModels = await _unitOfWork.LessonRepository.GetStudentInfoAsync(studentId);
 
-            return _mapper.Map<IList<StudentLessonDto>>(studentLessonModels) ?? null;
+            return Result<IList<StudentLessonDto>>.GetSuccess( _mapper.Map<IList<StudentLessonDto>>(studentLessonModels));
         }
```

Comment:
Add check for null. If null return Result<T>.Error()

Reply:
done